backend/src/gamestate/game.rs

```diff
@@ -0,0 +1,97 @@
+use crate::cards::deck::Deck;
+use crate::gamestate::player::Player;
+use crate::gamestate::serialization::{FinishedStatus, LobbyStatus, RunningStatus};
+use crate::gamestate::WSMessage;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone, Copy)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum GameStatus {
+    Lobby,
+    Running,
+    Finished,
+}
+
+pub struct Game {
+    status: GameStatus,
+    pub players: Vec<Player>,
+    deck: Deck,
+    turns_played: usize,
+}
+
+impl Game {
+    pub fn new(author_name: String) -> Game {
+        Game {
+            status: GameStatus::Lobby,
+            players: vec![Player::new(author_name, true)],
+            deck: Deck::new(),
+            turns_played: 0,
+        }
+    }
+
+    pub fn find_player(&self, name: String) -> Option<&Player> {
+        self.players
+            .iter()
+            .find(|player| player.get_name_clone() == name)
+    }
+
+    pub fn find_author(&self) -> Option<&Player> {
+        self.players.iter().find(|player| player.is_author)
+    }
+
+    pub fn find_author_name(&self) -> String {
+        match self.find_author() {
+            None => "UnknownAuthor".into(),
+            Some(author) => author.get_name_clone(),
+        }
+    }
```

Comment:
Nejsem si jistý, zda budeme `find_author_name()` někdy používat, ale to by zas tak nevadilo. Spíš takto má Game moc věcí, za které je zodpovědná (odpovědnost by se v tomto případě dala delegovat na hráče), takže pak může být zbytečně velká...

Je nějaký důvod, proč namísto funkce `find_author_name()` nepoužít `game.find_author().name()`?

Reply:
> zda budeme find_author_name() někdy používat

Však už ho používám, na struktury pro status WSMessages.

Reply:
Fixed by moving from game.rs Game struct to serialization.rs as private methods in c40a3d9bf47a721fe1889926b568ace2d1684e33.